subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/api/types.go

```diff
@@ -158,3 +158,51 @@ type LazyPreemptionStatus struct {
 	// It was lazy preempted at PreemptionTime.
 	PreemptionTime meta.Time `json:"preemptionTime"`
 }
+
+type ClusterStatus struct {
+	// Status of cells in the physical cluster
+	PhysicalCluster []*PhysicalCellStatus `json:"physicalCluster"`
+	// Status of cells in each VC
+	VirtualClusters map[string][]*VirtualCellStatus `json:"virtualClusters"`
+}
+
+type CellStatus struct {
+	GpuType  string `json:"gpuType,omitempty"`
+	CellType string `json:"cellType"`
+	// Address of a physical cell consists of its address (or index) in each level
+	// (e.g., node0/0/0/0 may represent node0, CPU socket 0, PCIe switch 0, GPU 0.
+	// Address of a virtual cell consists of its VC name, index of the preassigned cell,
+	// and the relative index in each level inside the preassigned cell
+	// (e.g., VC1/0/0 may represent VC1, preassigned cell 0, index 0 among its children)
+	CellAddress string `json:"cellAddress"`
+	// State can be one of "Free", "Used", and "Bad"
+	State    string `json:"state"`
+	Priority int32  `json:"priority"`
+}
+
+type PhysicalCellStatus struct {
+	CellStatus
+	Children    []*PhysicalCellStatus `json:"children,omitempty"`
+	Vc          string                `json:"vc,omitempty"`
+	VirtualCell *VirtualCellStatus    `json:"virtualCell,omitempty"`
```

Comment:
Move VC field into virtualCell? It can be redundant like gputype

Reply:
Where redundant? VirtualCell does not have VC field.
Also, a oppor physical cell does not have a VirtualCell field.